codex-rs/tui/src/resume_picker.rs

```diff
@@ -0,0 +1,453 @@
+        return;
+    }
+
+    // Compute how many rows fit (1 line per item)
+    let capacity = area.height as usize;
+    let start = state.selected.saturating_sub(capacity.saturating_sub(1));
+    let visible = &rows[start..rows.len().min(start + capacity)];
+
+    let mut y = area.y;
+    for (idx, row) in visible.iter().enumerate() {
+        let is_sel = start + idx == state.selected;
+        let marker = if is_sel { "> ".bold() } else { "  ".into() };
+        let ts = row
+            .ts
+            .map(human_time_ago)
+            .unwrap_or_else(|| "".to_string())
+            .dim();
+        let max_cols = area.width.saturating_sub(6) as usize;
+        let preview = truncate_text(&row.preview, max_cols);
+
+        let line: Line = vec![marker, ts, "  ".into(), preview.into()].into();
+        let rect = Rect::new(area.x, y, area.width, 1);
+        frame.render_widget_ref(line, rect);
+        y = y.saturating_add(1);
+    }
+}
+
+fn human_time_ago(ts: DateTime<Utc>) -> String {
+    let now = Utc::now();
+    let delta = now - ts;
+    let secs = delta.num_seconds();
+    if secs < 60 {
+        let n = secs.max(0);
+        if n == 1 {
+            format!("{n} second ago")
+        } else {
+            format!("{n} seconds ago")
+        }
+    } else if secs < 60 * 60 {
+        let m = secs / 60;
+        if m == 1 {
+            format!("{m} minute ago")
+        } else {
+            format!("{m} minutes ago")
+        }
+    } else if secs < 60 * 60 * 24 {
+        let h = secs / 3600;
+        if h == 1 {
+            format!("{h} hour ago")
+        } else {
+            format!("{h} hours ago")
+        }
+    } else {
+        let d = secs / (60 * 60 * 24);
+        if d == 1 {
+            format!("{d} day ago")
+        } else {
+            format!("{d} days ago")
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+
+    fn head_with_ts_and_user_text(ts: &str, texts: &[&str]) -> Vec<serde_json::Value> {
+        let mut head = vec![json!({"timestamp": ts})];
+        head.push(json!({
+            "type":"message",
+            "role":"user",
+            "content": texts.iter().map(|t| json!({"type":"input_text","text": *t})).collect::<Vec<_>>()
+        }));
+        head
+    }
+
+    #[test]
+    fn skips_user_instructions_and_env_context() {
+        let head = vec![
+            json!({"timestamp": "2025-01-01T00:00:00Z"}),
+            json!({
+                "type":"message","role":"user","content":[{"type":"input_text","text":"<user_instructions>hi</user_instructions>"}]
+            }),
+            json!({
+                "type":"message","role":"user","content":[{"type":"input_text","text":"<environment_context>cwd</environment_context>"}]
+            }),
+            json!({
+                "type":"message","role":"user","content":[{"type":"input_text","text":"real question"}]
+            }),
```

Comment:
Can you format these? `just fmt` doesn't do it because it does not format content inside a macro since whitespace could be significant.